src/main/java/com/jnape/palatable/lambda/monoid/builtin/MapMerge.java

```diff
@@ -0,0 +1,47 @@
+package com.jnape.palatable.lambda.monoid.builtin;
+
+import com.jnape.palatable.lambda.functions.Fn1;
+import com.jnape.palatable.lambda.functions.specialized.BiMonoidFactory;
+import com.jnape.palatable.lambda.functions.specialized.MonoidFactory;
+import com.jnape.palatable.lambda.monoid.Monoid;
+import com.jnape.palatable.lambda.semigroup.Semigroup;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static com.jnape.palatable.lambda.monoid.Monoid.monoid;
+
+public class MapMerge<A, B> implements BiMonoidFactory<Supplier<Map<A, B>>, Semigroup<B>, Map<A, B>> {
```

Comment:
Can we also use `K` and `V` for the parameters, given that we're talking about `j.u.Map` here.